gpii/node_modules/gpii-oauth2/gpii-oauth2-authz-server/public/src/components/privacySettings/js/ServicesMenu.js

```diff
@@ -0,0 +1,144 @@
+/*!
+Copyright 2015 OCAD University
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+The research leading to these results has received funding from the European Union's
+Seventh Framework Programme (FP7/2007-2013) under grant agreement no. 289016.
+
+You may obtain a copy of the License at
+https://github.com/GPII/universal/blob/master/LICENSE.txt
+*/
+
+// Declare dependencies
+/* global fluid, gpii, jQuery */
+
+(function ($, fluid) {
+
+    "use strict";
+
+    fluid.defaults("gpii.oauth2.servicesMenu", {
+        gradeNames: ["fluid.viewComponent"],
+        widgetOptions: {
+            select: "{that}.fireOnServiceSelected",
+            focus: "{that}.addMenuItemFocusCss"
+        },
+        events: {
+            onServiceSelected: null,
+            onClose: null
+        },
+        model: {
+            isMenuOpen: false
+        },
+        modelListeners: {
+            isMenuOpen: [{
+                "this": "{that}.container",
+                method: "toggleClass",
+                args: ["{that}.options.styles.menuOpen", "{change}.value"]
+            }, {
+                funcName: "gpii.oauth2.servicesMenu.handleMenuState",
+                args: [
+                    "{that}.container",
+                    "{that}.model.isMenuOpen",
+                    "{that}.close",
+                    "{that}.events.onClose"
+                ],
+                excludeSource: "init"
+            }]
+        },
+        listeners: {
+            "onCreate.setup": {
+                "this": "{that}.container",
+                method: "menu",
+                args: ["{that}.options.widgetOptions"]
+            },
+            "onCreate.bindEscape": {
+                "this": "{that}.container",
+                method: "keydown",
+                args: ["{that}.bindEscape"]
+            }
+        },
+        invokers: {
+            open: {
+                func: "{that}.setIsMenuOpen",
+                args: true
+            },
+            close: {
+                func: "{that}.setIsMenuOpen",
+                args: false
+            },
+            setIsMenuOpen: {
```

Comment:
I think we can do without this method in favour of manipulating the model directly